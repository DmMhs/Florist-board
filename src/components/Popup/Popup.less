.Popup {
  width: fit-content;
  padding: 8px;
  border: 1px solid #eee;
  border-radius: 5px;
  margin: 5px 0;
  i {
    font-size: 1.3rem;
    color: white;
    margin: 0 5px;
  }
  span {
    font-size: 1.3rem;
    color: white;
  }
}
