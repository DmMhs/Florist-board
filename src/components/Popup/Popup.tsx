import React from 'react';

import './Popup.less';

type PopupTypes = 'success' | 'failure' | 'info';

interface PopupProps {
  type: PopupTypes | null;
  message: string;
  key?: string;
}

const popup = (props: PopupProps) => {
  const { type, message } = props;
  let color: string;
  let icon;

  switch (type) {
    case 'success':
      icon = <i className="far fa-check-square" />;
      color = 'rgb(134, 218, 134)';
      break;
    case 'failure':
      icon = <i className="far fa-window-close" />;
      color = 'rgb(218, 69, 69)';
      break;
    case 'info':
      icon = <i className="fas fa-info-circle" />;
      color = 'skyblue';
      break;
    default:
      color = 'transparent';
  }

  return (
    <div
      className="Popup"
      style={{
        backgroundColor: color
      }}
    >
      <span className="popup-message">
        {icon}
        {message}
      </span>
    </div>
  );
};

export default popup;
