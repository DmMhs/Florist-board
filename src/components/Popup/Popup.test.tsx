import React from 'react';
import { shallow, mount } from 'enzyme';

import Popup from './Popup';

describe('Popup works as expected', () => {
  it('Popup component matches a snapshot', () => {
    const wrapper = shallow(
      <Popup type="success" message={`Product CORN was added to the cart`} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Type of component has impact on its styling', () => {
    let icon;
    let color;
    let wrapper = mount(<Popup type="success" message="wqewasdfas" />);
    const changeStyle = () => {
      if (wrapper.props().type === 'success') {
        icon = <i className="far fa-check-square" />;
        color = 'rgb(134, 218, 134)';
      } else if (wrapper.props().type === 'failure') {
        icon = <i className="far fa-window-close" />;
        color = 'rgb(218, 69, 69)';
      } else if (wrapper.props().type === 'info') {
        icon = <i className="fas fa-info-circle" />;
        color = 'skyblue';
      } else {
        color = 'transparent';
      }
    };
    changeStyle();
    expect(icon).toEqual(<i className="far fa-check-square" />);
    expect(color).toEqual('rgb(134, 218, 134)');
    wrapper = mount(<Popup type="failure" message="asfasfasf" />);
    changeStyle();
    expect(icon).toEqual(<i className="far fa-window-close" />);
    expect(color).toEqual('rgb(218, 69, 69)');
    wrapper = mount(<Popup type="info" message="gdsdfhr" />);
    changeStyle();
    expect(icon).toEqual(<i className="fas fa-info-circle" />);
    expect(color).toEqual('skyblue');
    wrapper = mount(<Popup type={null} message="gdsdfasfasfhr" />);
    changeStyle();
    expect(color).toEqual('transparent');
  });
});
