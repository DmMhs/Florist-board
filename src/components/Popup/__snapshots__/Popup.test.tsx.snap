// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup works as expected Popup component matches a snapshot 1`] = `
<div
  className="Popup"
  style={
    Object {
      "backgroundColor": "rgb(134, 218, 134)",
    }
  }
>
  <span
    className="popup-message"
  >
    <i
      className="far fa-check-square"
    />
    Product CORN was added to the cart
  </span>
</div>
`;
